### PR TITLE
feat: Api fetching using custom hook

### PR DESCRIPTION
**Pull request by PavanVaranasi2**
- added some stylings using MUI components and theme file
- also included custom hook file in hooks folder
- also made it responsive as well
- In the last image I provided an image to show that useEffect is now only rendering once with empty dependency array, and it is possible by removing React.StrictMode component from index.tsx, which always double-invokes certain lifecycle methods (including useEffect) to help identify side effects in the render phase.
- Also inlcuded some folders like atoms with .gitkeep file to make even that file to be tracked even without any files inside of it

Screenshot 2024-07-04 at 5 45 54 PM
Screenshot 2024-07-04 at 5 49 56 PM

Screenshot 2024-07-04 at 5 47 24 PM

### CODE REVIEW

File: src/components/molecules/PostCard.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import {
+  Card,
+  CardContent,
+  Typography as MuiTypography,
+  styled,
+} from "@mui/material";
+import { Post } from "../../hooks/useFetchPosts";
+
+interface PostCardProps {
```
Comment (MajaharZemoso) [inline src/components/molecules/PostCard.tsx]: There are no additional props beyond those in the `Post` interface. Therefore, there's no need to create another interface. Create a single interface file and reuse it in both files.

Comment (PavanVaranasi2) [reply src/components/molecules/PostCard.tsx:9]: ok got it, just a minute

**Review by MajaharZemoso — CHANGES_REQUESTED** [review]
Everything looks good to me except for the interface.

**Review by PavanVaranasi2 — COMMENTED** [review]
- removed redundant usage of interfaces in molecules and organisms

**Review by MajaharZemoso — APPROVED** [review]

**Pull request merged**